scripts/update-copyright.py

```diff
@@ -1,44 +1,74 @@
+# Copyright (c) 2024 Orange. All rights reserved.
+# This software is distributed under the BSD 3-Clause-clear License, the text of which is available
+# at https://spdx.org/licenses/BSD-3-Clause-Clear.html or see the "LICENSE" file for more details.
+
 """Updates the copyright notice of the input files"""
 
 import argparse
+import filecmp
 import os
+import tempfile
+import sys
+import shutil
 from datetime import datetime
 
 # pylint: disable=line-too-long
 byte_linesep = bytes(os.linesep, encoding="ascii")
 copyright_banner_lines = [
     bytes(
-        f"// Copyright (c) {datetime.today().year} Orange. All rights reserved.",
+        f"Copyright (c) {datetime.today().year} Orange. All rights reserved.",
         encoding="ascii",
     ),
-    b"// This software is distributed under the BSD 3-Clause-clear License, the text of which is available",
-    b'// at https://spdx.org/licenses/BSD-3-Clause-Clear.html or see the "LICENSE" file for more details.',
+    b"This software is distributed under the BSD 3-Clause-clear License, the text of which is available",
+    b'at https://spdx.org/licenses/BSD-3-Clause-Clear.html or see the "LICENSE" file for more details.',
 ]
 # pylint: enable=line-too-long
 
 
 def main(args):
     """Main method"""
+    # Process files and keep track if there were modifications
+    were_files_modified = False
     for file_path in args.file_paths:
-        update_copyright(file_path)
+        is_file_modified = update_copyright(file_path)
+        if is_file_modified:
+            were_files_modified = True
+            print(f"Updated {file_path}")
+
+    # Set the return code
+    return_code = 0
+    if were_files_modified:
+        return_code = 1
+
+    return return_code
 
 
 def update_copyright(file_path):
-    """Updates the copyright notice of a file"""
-    print(f"Updating {file_path}")
+    """Updates the copyright notice of a file if necessary"""
+
+    # Obtain the comment prefix string from the file extension
+    _, ext = os.path.splitext(file_path)
+    if ext in (".h", ".c", ".hpp", ".cpp", ".java"):
+        comment_prefix = b"// "
+    elif ext == ".py":
```

Comment:
Perhaps we should do the same for Bash scripts as well (`khiops-env`, `khiops`, `khiops_coclustering`) and their Windows counterparts?

Reply:
The issue stated that we'll do it only for `.py`. It was discussed in a weekly.

Reply:
Ok, I understand your point. However the issue (#150) states "Extend the copyright script for python and script files". So, I would either edit the issue description, or update the PR according to the issue.

Reply:
Done